presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -212,6 +241,20 @@ class EventViewModel @Inject constructor(
             is EventScreenEvent.CheckVisitorExists -> {
                 verifyVisitorEmail(eventScreenEvent.visitorEmail)
             }
+            is EventScreenEvent.CheckVisitorAlreadyAdded -> {
+                _eventScreenState.update { eventScreenState ->
+                    eventScreenState.copy(
+                        isAlreadyAdded = eventScreenEvent.isAlreadyAdded
+                    )
+                }
+            }
+            is EventScreenEvent.OnVerifyingVisitorEmail -> {
+                _eventScreenState.update { eventScreenState ->
+                    eventScreenState.copy(
+                        isVerifyingVisitorEmail = eventScreenEvent.isVerifyingVisitorEmail
+                    )
+                }
+            }
```

Comment:
This could all be just one method that get's called in the "AddVisitor"

ex:

fun validateVisitor()

You validate if the user is added or duplicate or whatever, and you can just return one error message if necessary

So you can avoid the when/if expression in the first comment I made